data/transactions/logic/eval.go

```diff
@@ -2102,6 +2103,56 @@ func opGtxnsa(cx *evalContext) {
 	cx.stack[last] = sv
 }
 
+func opGaidImpl(cx *evalContext, groupIdx int, opName string) (sv stackValue, err error) {
+	if groupIdx >= len(cx.TxnGroup) {
+		err = fmt.Errorf("%s lookup TxnGroup[%d] but it only has %d", opName, groupIdx, len(cx.TxnGroup))
+		return
+	} else if groupIdx > cx.GroupIndex {
+		err = fmt.Errorf("%s can't get future creatable ID of txn with index %d", opName, groupIdx)
```

Comment:
this message is not wrong, but could be better written:
%s can't get creatable ID of txn ahead of the current one (index %d) in the transaction group

I think that the wording of "future" is misleading, as it might refer to future transaction groups or future rounds.